backend/services/theme_service.py

```
@@ -0,0 +1,87 @@
+
+            logger.info(f"✅ Successfully generated and stored daily themed image at: {public_url}")
+            return public_url
+
+        except Exception as e:
+            logger.error(f"Failed to generate daily themed image: {e}", exc_info=True)
+            # REFRESH.MD: Raise a specific HTTPException that the router can handle.
+            raise HTTPException(status_code=500, detail="Failed to create the daily themed avatar image.") from e
+
+# --- FastAPI Dependency Injection ---
+def get_theme_service(
+    stability_service: StabilityAiService = Depends(get_stability_service),
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)